scipy/spatial/src/distance_wrap.c

```diff
@@ -32,6 +32,16 @@
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
+/* avoid unknown pragma warning in clang which may be used by developers */
+#ifndef __clang__
```

Comment:
+1, but actually [GCC 4.0](https://gcc.gnu.org/onlinedocs/gcc-4.0.4/gcc/Optimize-Options.html#Optimize-Options) already accepted all of these so `__GNUC__ >= 4` should be enough. (Not sure if GCC will complain about unknown options?)